lib/src/migrators/module.dart

```diff
@@ -389,10 +398,23 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
   /// it exists, or as a global variable otherwise.
   void _declareVariable(VariableDeclaration node) {
     if (_localScope == null || node.isGlobal) {
-      if (node.isGuarded &&
-          _globalVariables.containsKey(node.name) &&
+      if (_globalVariables.containsKey(node.name) &&
           _globalVariables[node.name].span.sourceUrl != _currentUrl) {
-        _configuredVariables.add(_globalVariables[node.name]);
+        if (node.isGuarded) {
+          _configuredVariables.add(_globalVariables[node.name]);
+        } else if (!_upstreamStylesheets
+            .contains(_globalVariables[node.name].span.sourceUrl)) {
```

Comment:
Consider putting `_globalVariables[node.name].span.sourceUrl` (and maybe also `_globalVariables[node.name]`) in a variable since it's long and referenced multiple times.